reflex/utils/exec.py

```diff
@@ -184,25 +193,46 @@ def run_backend(
         port: The app port
         loglevel: The log level.
     """
-    import uvicorn
-
-    config = get_config()
     app_module = f"reflex.app_module_for_backend:{constants.CompileVars.APP}"
 
     web_dir = get_web_dir()
     # Create a .nocompile file to skip compile for backend.
     if web_dir.exists():
         (web_dir / constants.NOCOMPILE_FILE).touch()
-
     # Run the backend in development mode.
-    uvicorn.run(
-        app=f"{app_module}.{constants.CompileVars.API}",
-        host=host,
-        port=port,
-        log_level=loglevel.value,
-        reload=True,
-        reload_dirs=[config.app_name],
-    )
+    if should_use_granian():
+        console.debug("Using Granian for backend")
+        try:
```

Comment:
We should move this block into its own function

Reply:
I can do that.